app/controllers/step_by_step_pages_controller.rb

```diff
@@ -76,9 +76,12 @@ def publish
   def schedule
     set_current_page_as_step_by_step
     if request.post?
-      if @step_by_step_page.update_attributes(scheduled_at: params[:scheduled_at])
+      date_params = params[:schedule][:date].permit(:year, :month, :day)
+      parser = DatetimeParser.new(date: date_params, time: params[:schedule][:time])
+      scheduled_at = parser.parse
```

Comment:
`parser.parse` returns `nil` if the date is invalid, but `@step_by_step_page.update_attributes(scheduled_at: scheduled_at)` will still be successful, that means a change note will be added for a `nil` date.

Reply:
Good spot - have changed the conditional.